### PR TITLE
Fix joint init data

### PR DESCRIPTION
**Pull request by papachap**
This bug fix includes changes to modules: `l_miter`, `t_step_joint`, `t_birdsmouth`, `t_dovetail`.

1.     Update __init__ to pass **kwargs to the superclass
2.     Update __data__ to include beams directly (rather than their GUIDs).
3.     Update restore_beams_from_keys to use model.element_by_guid() instead of model.elementsdict[]

### What type of change is this?

- [X] Bug fix in a **backwards-compatible** manner.
- [ ] New feature in a **backwards-compatible** manner.
- [ ] Breaking change: bug fix or new feature that involve incompatible API changes.
- [ ] Other (e.g. doc update, configuration, etc)

### Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [X] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [X] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas_timber.datastructures.Beam`.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/compas_timber/connections/l_miter.py [hunk 1]
```diff
@@ -41,13 +41,13 @@ class LMiterJoint(Joint):
     @property
     def __data__(self):
         data = super(LMiterJoint, self).__data__
-        data["beam_a"] = self.beam_a_guid
-        data["beam_b"] = self.beam_b_guid
+        data["beam_a"] = self.beam_a
```
Comment (chenkasirer) [inline src/compas_timber/connections/l_miter.py:44]: it is on purpose that guids are serialized here instead of the beams themselves. 
the beam objects are already serialized as part of the `Model` serialization. if these are assigned to `__data__` the same beam object might be serialized multiple times resulting in bloated files.

**Comment by papachap** [comment]
@chenkasirer  adjustments done 👍

File: CHANGELOG.md [hunk 2]
```diff
@@ -38,6 +38,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Renamed `intersection_line_plane` to `intersection_line_plane_param`.
 * Renamed `intersection_line_line_3D` to `intersection_line_line_param`.
 * Adjusted functions in `compas_timber._fabrication.DovetailMortise` and `compas_timber.connections.TDovetailJoint`.
+* Fixed `**kwargs` inheritance in `__init__` for joint modules: `LMiterJoint`, `TStepJoint`, `TDovetailJoint`, `TBirdsmouthJoint`.
+* Fixed GUID assignment logic from `**kwargs` to ensure correct fallback behavior for joint modules: `LMiterJoint`, `TStepJoint`, `TDovetailJoint`, `TBirdsmouthJoint`.
+* Changed `model.element_by_guid()` instead of direct `elementsdict[]` access for beam retrieval in joint modules: `LMiterJoint`, `TStepJoint`, `TDovetailJoint`, `TBirdsmouthJoint`.
+
```
Comment (chenkasirer) [inline CHANGELOG.md:44]: unnecessary whitespace
```suggestion
```

**Review by chenkasirer — APPROVED** [review]
LGTM!
